### PR TITLE
Update `dpnp.linalg.eig/eigvals`

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**
This PR proposes to update `dpnp.linalg.eig()` and `dpnp.linalg.eigvals()` functions using `numpy` call. 
Using `numpy.linalg.eig/eigvals` is a temporary solution until MKL implements `oneapi::mkl::lapack::geev` for USM interface.

This PR also removes `dpnp_algo_linalg.pyx` file because it no longer contains Cython functions

- [X] Have you provided a meaningful PR description?
- [X] Have you added a test, reproducer or referred to issue with a reproducer?
- [X] Have you tested your changes locally for CPU and GPU devices?
- [X] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

**Comment by vlad-perevezentsev** [comment]
@antonwolfy also I would also remove the backend implementation of `dpnp_eigvals_c` and `dpnp_eig_c` , because it is only implemented for symmetric arrays which is wrong.
What do you think about this?

**Comment by antonwolfy** [comment]
> @antonwolfy also I would also remove the backend implementation of `dpnp_eigvals_c` and `dpnp_eig_c` cause it is only implemented for symmetric arrays which is wrong. What do you think about this?

We definitely need to plan a clean up, but I would prefer to do that by a separate PR (like to remove all unused linalg function from backend and so on). But needs to be careful to avoid an unexpected impact on numba-dpex.

File: dpnp/linalg/dpnp_iface_linalg.py [hunk 1]
```diff
@@ -249,20 +246,101 @@ def det(a):
...
+    array([1.+1.j, 1.-1.j])
+    array([[0.70710678+0.j        , 0.70710678-0.j        ],
+           [0.        -0.70710678j, 0.        +0.70710678j]])
+
+    Complex-valued matrix with real e-values (but complex-valued e-vectors);
+    note that ``a.conj().T == a``, i.e., `a` is Hermitian.
+
+    >>> a = np.array([[1, 1j], [-1j, 1]])
+    >>> w, v = LA.eig(a)
+    >>> w; v
```
Comment (antonwolfy) [inline dpnp/linalg/dpnp_iface_linalg.py:315]:
```suggestion
    >>> w, v
```

File: dpnp/linalg/dpnp_iface_linalg.py [hunk 2]
```diff
@@ -249,20 +246,101 @@ def det(a):
...
+                              (conjugate symmetric) or a real symmetric matrix.
+    :obj:`dpnp.linalg.eigvalsh` : Compute the eigenvalues of a complex Hermitian or
+                                  real symmetric matrix.
+
+    Examples
+    --------
+    >>> import dpnp as np
+    >>> from dpnp import linalg as LA
+
+    (Almost) trivial example with real e-values and e-vectors.
```
Comment (antonwolfy) [inline dpnp/linalg/dpnp_iface_linalg.py:292]:
```suggestion
    (Almost) trivial example with real eigenvalues and eigenvectors.
```

File: dpnp/linalg/dpnp_iface_linalg.py [hunk 3]
```diff
@@ -249,20 +246,101 @@ def det(a):
...
+
+    >>> w, v = LA.eig(np.diag((1, 2, 3)))
+    >>> w; v
+    array([1., 2., 3.])
+    array([[1., 0., 0.],
+           [0., 1., 0.],
+           [0., 0., 1.]])
+
+    Real matrix possessing complex e-values and e-vectors; note that the
+    e-values are complex conjugates of each other.
```
Comment (antonwolfy) [inline dpnp/linalg/dpnp_iface_linalg.py:302]:
```suggestion
    Real matrix possessing complex eigenvalues and eigenvectors;
    note that the eigenvalues are complex conjugates of each other.
```

File: dpnp/linalg/dpnp_iface_linalg.py [hunk 4]
```diff
@@ -249,20 +246,101 @@ def det(a):
...
+
+    >>> w, v = LA.eig(np.array([[1, -1], [1, 1]]))
+    >>> w; v
+    array([1.+1.j, 1.-1.j])
+    array([[0.70710678+0.j        , 0.70710678-0.j        ],
+           [0.        -0.70710678j, 0.        +0.70710678j]])
+
+    Complex-valued matrix with real e-values (but complex-valued e-vectors);
+    note that ``a.conj().T == a``, i.e., `a` is Hermitian.
```
Comment (antonwolfy) [inline dpnp/linalg/dpnp_iface_linalg.py:311]:
```suggestion
    Complex-valued matrix with real eigenvalues (but complex-valued
    eigenvectors); note that ``a.conj().T == a``, i.e., `a` is Hermitian.
```

File: dpnp/linalg/dpnp_iface_linalg.py [hunk 5]
```diff
@@ -249,20 +246,101 @@ def det(a):
...
-            return dpnp_eig(x1_desc)
+    dpnp.check_supported_arrays_type(a)
+    check_stacked_2d(a)
+    check_stacked_square(a)
 
-    return call_origin(numpy.linalg.eig, x1)
+    a_sycl_queue = a.sycl_queue
+    a_usm_type = a.usm_type
+
+    w_np, v_np = numpy.linalg.eig(dpnp.asnumpy(a))
```
Comment (antonwolfy) [inline dpnp/linalg/dpnp_iface_linalg.py:339]: please add a comment why we are calling numpy here

File: dpnp/linalg/dpnp_iface_linalg.py [hunk 6]
```diff
@@ -334,32 +412,70 @@ def eigh(a, UPLO="L"):
...
-    x1_desc = dpnp.get_dpnp_descriptor(input, copy_when_nondefault_queue=False)
-    if x1_desc:
-        if x1_desc.size > 0:
-            return dpnp_eigvals(x1_desc).get_pyobj()
+    dpnp.check_supported_arrays_type(a)
+    check_stacked_2d(a)
+    check_stacked_square(a)
 
-    return call_origin(numpy.linalg.eigvals, input)
+    w_np = numpy.linalg.eigvals(dpnp.asnumpy(a))
```
Comment (antonwolfy) [inline dpnp/linalg/dpnp_iface_linalg.py:477]: please add a comment why we are calling numpy here

**Pull request merged**